pkg/csi/cinder/nodeserver.go

```diff
@@ -411,6 +411,30 @@ func (ns *nodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVol
 		}
 	}
 
+	// Try expanding the volume if it's created from a snapshot or another volume (see #1539)
+	if vol.SourceVolID != "" || vol.SnapshotID != "" {
+
+		r := mountutil.NewResizeFs(ns.Mount.Mounter().Exec)
+
+		needResize, err := r.NeedResize(devicePath, stagingTarget)
+
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not determine if volume %q need to be resized: %v", volumeID, err)
+		}
+
+		if needResize {
+			sourceId := vol.SourceVolID
+			if sourceId == "" {
+				sourceId = vol.SnapshotID
+			}
+			klog.V(4).Infof("NodeStageVolume: Resizing volume %q created from %q", volumeID, sourceId)
```

Comment:
I added a debug log and seems this time works fine to me now 👍 

```
I0607 10:23:25.219457       1 nodeserver.go:415] -------"3b0ff700-8d52-4be3-92cd-a46f9ae5e09f" ------ ""
I0607 10:23:25.222790       1 mount_linux.go:446] Attempting to determine if disk "/dev/disk/by-id/virtio-a8e4187e-473c-4e71-9" is formatted using blkid with args: ([-p -s TYPE -s PTTYPE -o export /dev/disk/by-id/virtio-a8e4187e-473c-4e71-9])
I0607 10:23:25.225982       1 mount_linux.go:449] Output: "DEVNAME=/dev/disk/by-id/virtio-a8e4187e-473c-4e71-9\nTYPE=ext4\n"
I0607 10:23:25.226046       1 resizefs_linux.go:123] ResizeFs.needResize - checking mounted volume /dev/disk/by-id/virtio-a8e4187e-473c-4e71-9
I0607 10:23:25.229287       1 nodeserver.go:425] -----resize--true
I0607 10:23:25.229306       1 nodeserver.go:432] NodeStageVolume: Resizing volume "a8e4187e-473c-4e71-9024-93eeb150239f" created from "3b0ff700-8d52-4be3-92cd-a46f9ae5e09f"
I0607 10:23:25.229328       1 mount_linux.go:446] Attempting to determine if disk "/dev/disk/by-id/virtio-a8e4187e-473c-4e71-9" is formatted using blkid with args: ([-p -s TYPE -s PTTYPE -o export /dev/disk/by-id/virtio-a8e4187e-473c-4e71-9])
I0607 10:23:25.232007       1 mount_linux.go:449] Output: "DEVNAME=/dev/disk/by-id/virtio-a8e4187e-473c-4e71-9\nTYPE=ext4\n"
I0607 10:23:25.232031       1 resizefs_linux.go:55] ResizeFS.Resize - Expanding mounted volume /dev/disk/by-id/virtio-a8e4187e-473c-4e71-9
I0607 10:23:25.295898       1 resizefs_linux.go:70] Device /dev/disk/by-id/virtio-a8e4187e-473c-4e71-9 resized successfully
```

if we can add 
`Resizing volume "a8e4187e-473c-4e71-9024-93eeb150239f" created from "3b0ff700-8d52-4be3-92cd-a46f9ae5e09f"`
what is 3b0ff7xxx (source volume or snapshot) it will be much helpful

Reply:
Great, I've changed the logs to these:

- Snapshot: `Resizing volume "<vol-id>" created from snapshot "<snapshot-id>"`
- Volume: `Resizing volume "<vol-id>" created from source volume "<source-vol-id>"`